### DISCUSSION TITLE
Updating Kimai2 from 1.10.2 to 1.12

### DISCUSSION THREAD

**Discussion by swgithu** [kimai/kimai#2295] [General]
Hi,

i have a problem updating my kimai2 installation from 1.10.2 to 1.12
I followed the [Update Guide ](https://www.kimai.org/documentation/updates.html)
1. git fetch --tags  -> Works great. 
 OUTPUT:
`remote: Enumerating objects: 4555, done.
remote: Counting objects: 100% (2671/2671), done.
remote: Compressing objects: 100% (455/455), done.
remote: Total 1426 (delta 1161), reused 1162 (delta 964), pack-reused 0
Receiving objects: 100% (1426/1426), 1.50 MiB | 3.72 MiB/s, done.
Resolving deltas: 100% (1161/1161), completed with 348 local objects.
From https://github.com/kevinpapst/kimai2
 * [new tag]           1.11       -> 1.11
 * [new tag]           1.11.1     -> 1.11.1
 * [new tag]           1.12       -> 1.12
`
2. git checkout 1.12 -> Fails 

_error: Your local changes to the following files would be overwritten by checkout:
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/workflows/coverage.yaml
.github/workflows/linting.yaml
.github/workflows/testing.yaml
.github_changelog_generator
CHANGELOG.md
SECURITY.md
UPGRADING.md
assets/app.js
assets/calendar.js
assets/js/KimaiPlugin.js
assets/js/KimaiTranslation.js
assets/js/plugins/KimaiAPI.js
assets/js/plugins/KimaiAPILink.js
assets/js/plugins/KimaiAlert.js
assets/js/plugins/KimaiAutocomplete.js
assets/js/plugins/KimaiConfirmationLink.js
assets/js/plugins/KimaiDatatable.js
assets/js/plugins/KimaiDatePicker.js
assets/js/plugins/KimaiDateRangePicker.js
assets/js/plugins/KimaiDateTimePicker.js
assets/js/plugins/KimaiFormSelect.js
assets/js/plugins/KimaiRecentActivities.js
assets/js/plugins/KimaiReducedClickHandler.js
assets/js/plugins/KimaiThemeInitializer.js
assets/sass/app.scss
assets/sass/calendar.scss
assets/sass/invoice-pdf.scss
assets/sass/sweetalert.scss
assets/sass/variables.scss
composer.json
composer.lock
config/bundles.php
config/packages/doctrine.yaml
config/packages/fos_rest.yaml
config/packages/kimai.yaml
config/packages/nelmio_api_doc.yaml
config/packages/twig.yaml
config/services.yaml
config/services_test.yaml
package.json
public/build/1.512c747c.js
public/build/app.554717a4.css
public/build/app.cafc411b.js
public/build/calendar.0e49d4da.js
public/build/entrypoints.json
public/build/invoice-pdf.340721a4.css
public/build/manifest.json
src/API/ActivityController.php
src/API/BaseApiController.php
src/API/ConfigurationController.php
src/API/CustomerController.php
src/API/Model/I18nConfig.php
src/API/ProjectController.php
src/API/TagController.php
src/API/TeamController.php
src/API/TimesheetController.php
src/API/UserController.php
src/Calendar/Google.php
src/Calendar/Source.php
src/Command/ImportCustomerCommand.php
src/Command/ImportTimesheetCommand.php
src/Command/InvoiceCreateCommand.php
src/Command/ReloadCommand.php
src/Command/ResetCommand.php
src/Configuration/CalendarConfiguration.php
src/Configuration/FormConfiguration.php
src/Configuration/StringAccessibleConfigTrait.php
src/Configuration/SystemConfiguration.php
src/Configuration/ThemeConfiguration.php
src/Configuration/TimesheetConfiguration.php
src/Constants.php
src/Controller/AboutController.php
src/Controller/AbstractController.php
src/Controller/ActivityController.php
src/Controller/CalendarController.php
src/Controller/CustomerController.php
src/Controller/DoctorController.php
src/Controller/ExportController.php
src/Controller/HomepageController.php
src/Controller/InvoiceController.php
src/Controller/PermissionController.php
src/Controller/ProfileController.php
src/Controller/ProjectController.php
src/Controller/SystemConfigurationController.php
src/Controller/TagController.php
src/Controller/TeamController.php
src/Controller/TimesheetAbstractController.php
src/Controller/TimesheetTeamController.php
src/Controller/UserController.php
src/DataFixtures/CustomerFixtures.php
src/DataFixtures/TeamFixtures.php
src/DataFixtures/TimesheetFixtures.php
src/DataFixtures/UserFixtures.php
src/DependencyInjection/Compiler/DoctrineCompilerPass.php
src/DependencyInjection/Compiler/ExportServiceCompilerPass.php
src/DependencyInjection/Configuration.php
src/Entity/ColorTrait.php
src/Entity/Customer.php
src/Entity/Invoice.php
src/Entity/MetaTableTypeTrait.php
src/Entity/Role.php
src/Entity/Timesheet.php
src/Entity/User.php
src/Entity/UserPreference.php
src/EventSubscriber/MenuSubscriber.php
src/EventSubscriber/UserPreferenceSubscriber.php
src/Export/Base/AbstractSpreadsheetRenderer.php
src/Export/Base/CsvRenderer.php
src/Export/Base/RendererTrait.php
src/Export/Base/XlsxRenderer.php
src/Export/ServiceExport.php
src/Export/Spreadsheet/Extractor/AnnotationExtractor.php
src/Export/Spreadsheet/Writer/XlsxWriter.php
src/Form/API/ActivityApiEditForm.php
src/Form/API/ProjectApiEditForm.php
src/Form/ActivityEditForm
Aborting_

I've tried the git rest --hard command but that doesn't help.
Output for git log --oneline --graph after I restored my backup
_* b69ea064 (grafted, HEAD, tag: 1.10.2) bugfix release 1.10.2 (#1942)_

I have not much experience in using github but i think the problem is that i have modified files which different from the repository. 

Would be great if someone can help me here... i'm a little bit lost^^

Best regards
Stefan

**Comment by kevinpapst** [comment]
Did you really change all these files? 
What does `git status` report now?
Did you retry the upgrade after the `git reset`?

**Reply by swgithu** [reply]
At least not intentionally. I have only installed one other plugin so far.
git status output:
[output_git_status.txt](https://github.com/kevinpapst/kimai2/files/5862394/output_git_status.txt)
I tried this command that i found in the guide: bin/console kimai:update

**Reply by kevinpapst** [reply]
You can normally use `git reset --hard` (which might remove intentional changes as well).
If that doesn't help, then install Kimai fresh, copy over you changed files (.en, plugin, invoices...) and run the updater.

This is not a Kimai problem, you might want to try to ask in a specialized community to get help for that git issue.

**Reply by swgithu** [reply]
ok that did not solve the problem unfortunately.
I have now tried to create a new kimai server.
I installed the latest version and then tried to import the data.
I read in another thread that it should be possible to migrate with this command also from v2 to a newer v2...
`bin/console kimai:import-v1 "mysql://DATABASEUSER:DATABASEPW@HOST_OLD:3307/kimai2?charset=utf8" "kimai2_" "kimai123" "DE" "EUR" --timezone="Europe/Amsterdam" --language="de"`
![image](https://user-images.githubusercontent.com/77892655/105903250-efe5ce80-601f-11eb-8ac8-a6cf211e3d42.png)

Am I completely wrong here or do you have a tip how I could import the data?
Or do i need to install kimai over the existing installation and then run the update again?

**Reply by kevinpapst** [reply]
You are posting random commands, instead of going through the process step by step and sharing results. Nobody can help you if you are jumping around. 

> I read in another thread that it should be possible to migrate with this command also from v2 to a newer v2...

Completely wrong.

**Reply by swgithu** [reply]
no i just thought that you know that command from you project....
i think i just managed it on myself by just creating a dump of the old database and importing it to the new database on the new server. Sry for the confussion i maybe created here... i'am new to kimai and git and was struggling with it. after 4 hours i think i finally made it...

Thanks anyway for your help. The project is really good and we really enjoy the time tracking with it.

**Reply by kevinpapst** [reply]
4 hours and a new server sounds ugly. An update should take max 30 minutes (normally faster, but assuming you read the docs line by line and type slowly 😁).

Something went awfully wrong with git, we just don't know what ... and it's terrible hard to help, if you don't see what happened.

Sorry for the troubles you had - really glad you solved the mystery!
